java/pattern/schema/Rule.java

```diff
@@ -127,6 +129,11 @@ private static void validateThen(String label, @Nullable Conjunction<? extends P
             throw GraqlException.of(INVALID_RULE_THEN.message(label, then));
         }
 
+        // rule 'has' cannot assign both an attribute type and a variable
+        if (numConstraints == 1 && then.has().size()==1 && then.has().get(0).type().isNamed() && then.has().get(0).attribute().isVariable()) {
```

Comment:
lets be consistent with spacing between the `==`. In one case you have spaces on either side, in the other you don't. It's a minor habit, bit if we're through and clean in execution we make our lives easier everywhere else :)

Reply:
Done